ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ethernet-interfaces.j2

```diff
@@ -10,17 +10,40 @@
 | --------- | ----------- | ---- | ----- | ----------- | ----------- | ------------- |
 {%     for ethernet_interface in ethernet_interfaces | arista.avd.convert_dicts('name') | arista.avd.natural_sort('name') %}
 {%         if ethernet_interface.channel_group.id is arista.avd.defined %}
-{%             set port_channel_interface = 'Port-Channel' + ethernet_interface.channel_group.id | string %}
-{%             if port_channel_interfaces[port_channel_interface].type is not defined or port_channel_interfaces[port_channel_interface].type not in ['routed', 'l3dot1q', 'l2dot1q'] %}
-{%                 set description = ethernet_interface.description | arista.avd.default("-") %}
-{%                 set mode = port_channel_interfaces[port_channel_interface].mode | arista.avd.default("access") %}
-{%                 set vlans = port_channel_interfaces[port_channel_interface].vlans | arista.avd.default('-') %}
-{%                 set native_vlan = port_channel_interfaces[port_channel_interface].native_vlan | arista.avd.default('-') %}
+{%             set port_channel_interface_name = 'Port-Channel' ~ ethernet_interface.channel_group.id %}
+{%             if port_channel_interfaces is arista.avd.defined %}
+{%                 for port_channel_interface in port_channel_interfaces | arista.avd.convert_dicts('name') %}
+{%                     if port_channel_interface_name == port_channel_interface.name  %}
+{%                         if port_channel_interface.type is not defined or port_channel_interface.type not in ['routed', 'l3dot1q', 'l2dot1q'] %}
+{%                             set description = ethernet_interface.description | default("-") %}
+{%                             set mode = port_channel_interface.mode | default("access") %}
+{%                             set vlans = port_channel_interface.vlans | default ('-') %}
+{%                             set native_vlan = port_channel_interface.native_vlan | default ('-') %}
+{%                             set channel_group = ethernet_interface.channel_group.id %}
+{%                             if port_channel_interface.trunk_groups is arista.avd.defined %}
+{%                                 set l2 = namespace() %}
+{%                                 set l2.trunk_groups = [] %}
+{%                                 for trunk_group in port_channel_interface.trunk_groups | arista.avd.natural_sort %}
+{%                                     do l2.trunk_groups.append(trunk_group) %}
+{%                                 endfor %}
+{%                             else %}
+{%                                 set l2 = namespace() %}
+{%                                 set l2.trunk_groups = "-" %}
+{%                             endif %}
+| {{ ethernet_interface.name }} | {{ description }} | *{{ mode }} | *{{ vlans }} | *{{ native_vlan }} | *{{ l2.trunk_groups }} | {{ channel_group }} |
+{%                         endif %}
+{%                     endif %}
+{%                 endfor %}
+{%             else %}
```

Comment:
This is an else condition for ```port_channel_interfaces``` is not defined. So don't access ```port_channel_interface.<>``` because its not defined.